modules/database/src/std/filters/ts.c

```diff
@@ -56,34 +160,207 @@ static db_field_log* filter(void* pvt, dbChannel *chan, db_field_log *pfl) {
     return pfl;
 }
 
-static void channelRegisterPre(dbChannel *chan, void *pvt,
-                               chPostEventFunc **cb_out, void **arg_out, db_field_log *probe)
-{
+static db_field_log *replace_fl_value(tsPrivate const *pvt,
+                                      db_field_log *pfl,
+                                      void (*func)(tsPrivate const *,
+                                                   db_field_log *)) {
+    /* Get rid of the old value */
+    if (pfl->type == dbfl_type_ref && pfl->u.r.dtor) {
+        pfl->u.r.dtor(pfl);
```

Comment:
It may only be paranoia, but can you add `pfl->u.r.dtor = NULL;` after running the old dtor.